File: backend/app/src/db/models/Graphic.ts
import {
  AutoIncrement,
  BelongsTo,
  Column,
  CreatedAt,
  DataType,
  ForeignKey,
  HasMany,
  Model,
  PrimaryKey,
  Table,
} from "sequelize-typescript";
import Item from './Item'
import Layout from "./LayoutTable";
import LayoutTable from "./LayoutTable";

@Table({ tableName: "Graphic" })
class Graphic extends Model<GraphicAttributes> {

  @PrimaryKey
  @AutoIncrement
  @Column(DataType.INTEGER)
  declare id: number;

  @Column(DataType.STRING)
  declare name: string;

  @Column(DataType.STRING)
  declare chartType: string;

  @Column(DataType.INTEGER)
  declare width: number;

  @Column(DataType.INTEGER)
  declare height: number;

  @HasMany(() => Item)
  declare itens: Item[];

  @ForeignKey(() => LayoutTable)
  @Column(DataType.INTEGER)
  declare layoutId: number;

  @BelongsTo(() => LayoutTable)
  declare layout: LayoutTable;
}

interface GraphicAttributes {
  id: number;
  name: string;
  width: number;
  height: number;
  chartType: string;
  itens: Item[];
  layoutId: number;
}

export default Graphic;

File: backend2/models/graphic/graphic.py
from sqlalchemy import ForeignKey
from sqlalchemy.orm import relationship
from db import db

class Graphic(db.Model):
    """
    Graphic model representing chart configurations.
    """
    __tablename__ = "graphics"

    # Primary Key
    id = db.Column(db.Integer, primary_key=True, autoincrement=True)

    # Columns
    name = db.Column(db.String, nullable=False)
    chart_type = db.Column(db.String, nullable=False)
    width = db.Column(db.Integer, nullable=False)
    height = db.Column(db.Integer, nullable=False)

    # ForeignKey
    layout_id = db.Column(db.Integer, ForeignKey("layout_tables.id"), nullable=False)

    # Relationship to LayoutTable
    layout = relationship("LayoutTable", back_populates="graphics")

    def to_dict(self):
        """
        Convert the model instance to a dictionary.
        """
        return {
            "id": self.id,
            "name": self.name,
            "chart_type": self.chart_type,
            "width": self.width,
            "height": self.height,
            "layout_id": self.layout_id,
        }
